src/Provider/PhoneNumberProvider.php

```diff
@@ -31,61 +22,36 @@ class PhoneNumberProvider implements PhoneNumberProviderInterface {
   protected $smsProvider;
 
   /**
-   * Storage for phone number settings.
+   * The phone number verification service.
    *
-   * @var \Drupal\Core\Config\Entity\ConfigEntityStorageInterface
+   * @var \Drupal\sms\Provider\PhoneNumberVerificationInterface
    */
-  protected $phoneNumberSettings;
-
-  /**
-   * Storage for Phone Number Verification entities.
-   *
-   * @var \Drupal\Core\Entity\EntityStorageInterface
-   */
-  protected $phoneNumberVerificationStorage;
-
-  /**
-   * The token service.
-   *
-   * @var \Drupal\Core\Utility\Token
-   */
-  protected $token;
-
-  /**
-   * The config factory.
-   *
-   * @var \Drupal\Core\Config\ConfigFactoryInterface
-   */
-  protected $configFactory;
+  protected $phoneNumberVerification;
 
   /**
    * Constructs a new PhoneNumberProvider object.
    *
-   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
-   *   The entity type manager.
-   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
-   *   The config factory.
-   * @param \Drupal\Core\Utility\Token $token
-   *   The token replacement system.
    * @param \Drupal\sms\Provider\SmsProviderInterface $sms_provider
    *   The SMS provider.
+   * @param \Drupal\sms\Provider\PhoneNumberVerificationInterface $phone_number_verification
+   *   The phone number verification service.
    */
-  public function __construct(EntityTypeManagerInterface $entity_type_manager, ConfigFactoryInterface $config_factory, Token $token, SmsProviderInterface $sms_provider) {
+  public function __construct(SmsProviderInterface $sms_provider, PhoneNumberVerificationInterface $phone_number_verification) {
     $this->smsProvider = $sms_provider;
-    $this->phoneNumberSettings = $entity_type_manager
-      ->getStorage('phone_number_settings');
-    $this->phoneNumberVerificationStorage = $entity_type_manager
-      ->getStorage('sms_phone_number_verification');
-    $this->token = $token;
-    $this->configFactory = $config_factory;
+    // Temporarily inject service until an event is created.
+    // See: https://www.drupal.org/node/2797121
+    $this->phoneNumberVerification = $phone_number_verification;
```

Comment:
So long as no-one overrides the service It'll be fine. I anticipate this will be removed soon, as soon as the discussed event in 2797121 is implemented.